Applications/Spire/Include/Spire/Ui/IconButton.hpp

```diff
@@ -22,48 +22,41 @@ namespace Spire {
       */
       explicit IconButton(QImage icon, QWidget* parent = nullptr);
 
-      //! Constructs an IconButton with a default icon and a hover icon.
+      //! Sets the icon to display.
       /*!
-        \param icon The icon shown when the button is not hovered.
-        \param hover_icon The icon shown when the button is hovered.
-        \param parent The parent QWidget to the IconButton.
+        \param icon The icon image.
       */
-      IconButton(QImage icon, QImage hover_icon, QWidget* parent = nullptr);
+      void set_icon(const QImage& icon);
 
-      //! Constructs an IconButton with a default icon, hover icon, and
-      //!  blur icon.
+      //! Sets the icon's color when the window lacks focus.
       /*!
-        \param icon The icon shown when the button is not hovered.
-        \param hover_icon The icon shown when the button is hovered.
-        \param blur_icon The icon shown when the window lacks focus.
-        \param parent The parent QWidget to the IconButton.
+        \param color The blur color.
       */
-      IconButton(QImage icon, QImage hover_icon, QImage blur_icon,
-        QWidget* parent = nullptr);
+      void set_blur_color(const QColor& color);
 
-      //! Sets the default stylesheet for the button.
+      //! Sets the icon's default color.
       /*!
-        \param stylesheet The default stylesheet.
+        \param color The default color.
       */
-      void set_default_style(const QString& stylesheet);
+      void set_default_color(const QColor& color);
 
-      //! Sets the stylesheet for when the button is hovered.
+      //! Sets the icon's hover color.
       /*!
-        \param stylesheet The hover stylesheet.
+        \param color The hover color.
       */
-      void set_hover_style(const QString& stylesheet);
-
-      //! Returns the icon displayed.
-      const QImage& get_icon() const;
-
-      //! Sets the icon to display.
-      void set_icon(QImage icon);
+      void set_hover_color(const QColor& color);
```

Comment:
Minimize the number of setters, instead have a constructor that takes these values and even better is to have a Styles struct that defines these along with sensible default values.

    struct Properties {
      QColor m_backgroundColor;
      QColor m_hoverColor;
      QColor m_blurColor;
    };